### PR TITLE
Add handy properties

### PR DESCRIPTION
**Pull request by LiangJYu**
This PR adds the following handy properties to `Sentinel1BurstSlc` class:
* sensing mid and stop as properties to be consistent with sensing start.
* length and width broken out of shape attribute

### CODE REVIEW

File: src/s1reader/s1_burst_slc.py [hunk 1]
```diff
@@ -313,6 +313,17 @@ def get_sensing_mid(self):
...
 
+    def get_sensing_end(self):
+        '''Returns sensing end as datetime.datetime object.
+
+        Returns:
+        --------
+        _ : datetime.datetime
+            Sensing end as datetime.datetime object.
+        '''
+        d_seconds = (self.shape[0] - 1) * self.azimuth_time_interval
```
Comment (hfattahi) [inline src/s1reader/s1_burst_slc.py:324]: Now that you are adding handy parameters, another handy parameter is length and width. :)

**Review by hfattahi — APPROVED** [review]

File: src/s1reader/s1_burst_slc.py [hunk 2]
```diff
@@ -313,6 +313,17 @@ def get_sensing_mid(self):
         d_seconds = 0.5 * (self.shape[0] - 1) * self.azimuth_time_interval
         return self.sensing_start + datetime.timedelta(seconds=d_seconds)
 
+    def get_sensing_end(self):
```
Comment (yunjunz) [inline src/s1reader/s1_burst_slc.py:316]: Should we call this `get_sensing_stop()`? It matches with `sensing_start`..

File: src/s1reader/s1_burst_slc.py [hunk 3]
```diff
@@ -431,3 +420,35 @@ def as_dict(self):
...
+    @property
+    def sensing_stop(self):
+        '''Returns sensing end as datetime.datetime object.
+
+        Returns:
+        --------
+        _ : datetime.datetime
+            Sensing end as datetime.datetime object.
+        '''
+        d_seconds = (self.shape[0] - 1) * self.azimuth_time_interval
```
Comment (yunjunz) [inline src/s1reader/s1_burst_slc.py:445]: Could you help understand why there is `- 1` here for both sensing_stop and sensing_mid?

Comment (LiangJYu) [reply src/s1reader/s1_burst_slc.py:445]: The `- 1` accounts for the first element. Without the `- 1`, the `sensing_stop` returns a time value outside radar grid.

Comment (hfattahi) [reply src/s1reader/s1_burst_slc.py]: makes sense for stop but not for mid.

File: src/s1reader/s1_burst_slc.py [hunk 4]
```diff
@@ -431,3 +420,35 @@ def as_dict(self):
...
+    @property
+    def sensing_mid(self):
+        '''Returns sensing mid as datetime.datetime object.
+
+        Returns:
+        --------
+        _ : datetime.datetime
+            Sensing mid as datetime.datetime object.
+        '''
+        d_seconds = 0.5 * (self.shape[0] - 1) * self.azimuth_time_interval
```
Comment (hfattahi) [inline src/s1reader/s1_burst_slc.py]:
```suggestion
        d_seconds = 0.5 * self.shape[0] * self.azimuth_time_interval
```

File: src/s1reader/s1_burst_slc.py (hunk 3 above)
Comment (yunjunz) [reply src/s1reader/s1_burst_slc.py]: For `sensing_stop`, I feel like this is similar to "a pixel is an area or a point" topic, if `sensing_start` is the beginning of the first pixel and `sensing_stop` is the end of the last pixel, then we should not use `- 1`, am I understanding it right?

@LiangJYu could you maybe show the value of the sensing_stop and radar grid boundaries here for us to better understand?

Comment (LiangJYu) [reply src/s1reader/s1_burst_slc.py]: image
[compute sensing stop.xlsx](https://github.com/opera-adt/s1-reader/files/8575769/compute.sensing.stop.xlsx)

Comment (hfattahi) [reply src/s1reader/s1_burst_slc.py]: @yunjunz sensing_stop here is actually the time-tag of the last sample. This is just the conversion between index and time. If sensing_start represents the beginning of the first pixel, sensing stop represents the beginning of the last pixel.

Comment (yunjunz) [reply src/s1reader/s1_burst_slc.py]: Thank you @LiangJYu and @hfattahi for the explanation.

**Review by yunjunz — APPROVED** [review]
Looks great to me.

**Pull request merged**